src/anndata/_core/xarray.py

```diff
@@ -195,4 +275,79 @@ def __setitem__(self, key, value):
         else:
             # maintain setting behavior of a 2D dataframe i.e., one dim
             value = (self.index_dim, value)
-        super().__setitem__(key, value)
+        self.ds.__setitem__(key, value)
+
+    def copy(self, *args, **kwargs):
+        """
+        Return a copy of the Dataset2D object.
+        """
+        as_2d = Dataset2D(self.ds.copy(*args, **kwargs))
+        as_2d.true_index_dim = self.true_index_dim
+        as_2d.is_backed = self.is_backed
+        return as_2d
+
+    def isel(self, *args, **kwargs):
+        """
+        Return a isel of the Dataset2D object.
+        """
+        as_2d = Dataset2D(self.ds.isel(*args, **kwargs))
+        as_2d.true_index_dim = self.true_index_dim
+        as_2d.is_backed = self.is_backed
+        return as_2d
+
+    def __iter__(self) -> Iterator[Hashable]:
+        return iter(self.ds)
+
+    def __len__(self) -> int:
+        return len(self.ds)
+
+    @property
+    def dtypes(self) -> pd.Series:
+        """
+        Return a Series with the dtypes of the variables in the Dataset2D.
+        """
+        return self.ds.dtypes
+
+    def equals(self, b: object) -> bool:
```

Comment:
This doesn't have a docstring, is this not supposed to be a public API? (ditto for `to_memory`)